### PR TITLE
eds_speed_test: add GrpcMuxImpl config ingestion path to benchmark.

### PR DESCRIPTION
**Pull request by htuch**
A lot of the overhead we see in
https://github.com/envoyproxy/envoy/issues/11362 has to do with the
overhead from the GrpcMuxImpl ingestion path, and helps put the EDS
cluster overhead in context.

This patch adds a GrpcMuxImpl and GrpcSubscription to the test,
allowing a fairly realistic cost of EDS update to be obtained.

Below is the example output. We're seeing v3 costs ~3.2x of the v2
ingestion. Future patches will narrow this.

---------------------------------------------------------------------------------
Benchmark                                       Time             CPU
Iterations
---------------------------------------------------------------------------------
priorityAndLocalityWeighted/0/0/2000      6277063 ns      6276456 ns
113
priorityAndLocalityWeighted/1/0/2000     20236878 ns     20236505 ns
35
priorityAndLocalityWeighted/0/1/2000      4589757 ns      4588833 ns
154
priorityAndLocalityWeighted/1/1/2000     14853274 ns     14850006 ns
42
priorityAndLocalityWeighted/0/0/4096     11865838 ns     11865314 ns
59
priorityAndLocalityWeighted/1/0/4096     39787876 ns     39787716 ns
18
priorityAndLocalityWeighted/0/1/4096      8186874 ns      8185724 ns
87
priorityAndLocalityWeighted/1/1/4096     29376139 ns     29374980 ns
24
priorityAndLocalityWeighted/0/0/32768   110248743 ns    110246101 ns
6
priorityAndLocalityWeighted/1/0/32768   343826306 ns    343761359 ns
2
priorityAndLocalityWeighted/0/1/32768    76962915 ns     76953406 ns
7
priorityAndLocalityWeighted/1/1/32768   254426231 ns    254401890 ns
3
priorityAndLocalityWeighted/0/0/100000  352576987 ns    352550994 ns
2
priorityAndLocalityWeighted/1/0/100000 1100804152 ns   1100646263 ns
1
priorityAndLocalityWeighted/0/1/100000  253870688 ns    253857626 ns
3
priorityAndLocalityWeighted/1/1/100000  808080918 ns    808028852 ns
1

Risk level: Low (test only)
Testing: Added ASSERTs to validate that we're loaded all the hosts at
  the end of the benchmark.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

**Comment by htuch** [comment]
CC @pgenera

File: test/common/upstream/eds_speed_test.cc [hunk 1]
```diff
@@ -30,7 +32,18 @@ namespace Upstream {
...
 public:
-  EdsSpeedTest(benchmark::State& state) : state_(state), api_(Api::createApiForTest(stats_)) {}
+  EdsSpeedTest(benchmark::State& state, bool v2_config)
+      : state_(state), v2_config_(v2_config),
+        type_url_(v2_config_
+                      ? "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment"
+                      : "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment"),
+        subscription_stats_(Config::Utility::generateStats(stats_)),
+        api_(Api::createApiForTest(stats_)), async_client_(new Grpc::MockAsyncClient()),
+        grpc_mux_(new Config::GrpcMuxImpl(
```
Comment (jmarantz) [inline test/common/upstream/eds_speed_test.cc:42]: std::make_shared?

Comment (htuch) [reply test/common/upstream/eds_speed_test.cc:42]: Sure, out of curiosity, what is the performance difference here?

Comment (jmarantz) [reply test/common/upstream/eds_speed_test.cc:42]: probably makes it slower if anything, but more conformant to modern style, no?

**Pull request merged**